Game.js

```diff
@@ -1,385 +1,871 @@
+  lines.push(
+    Player.health, Player.healthVolume, gun.activeOnes,
+    gun.maxCount, projector.activeOnes, projector.maxCount
+  );
+
+  for (let i = 0; i < lines.length; i++) {
+    const name = rectsName[i];
+    rects[name].parametr = lines[i];
+  }
+}
+
+function zombiUpdate() {
+  if (randomInteger(0, 10000) > 9900) {
+    const newZombi = addZombi();
+    if (newZombi) {
+      zombies.push(...newZombi);
+    }
+  }
+
+  for (const zomb of zombies) {
+
+    zomb.update();
+
+    if (!zomb.egg && zomb.y >= maxY) {
+      eggs.push(
+        new Egg(zomb.x +  zomb.image.width / 2,
+          zomb.y + zomb.image.height, images.egg)
+      );
+      zomb.egg = eggs[eggs.length - 1];
+      zombiAttak(zomb);
+    }
+
+    if (zomb.isDraw) {
+      ctx.drawImage(
+        zomb.image, 0, 0,
+        zomb.image.width, zomb.image.height,
+        zomb.x, zomb.y, zomb.image.width,
+        zomb.image.height
+      );
+    }
+  }
+}
+
+function chekZombi(zomb) {
+  return !zomb.isDraw;
+}
+
+function weaponUpdate() {
+  for (const bul of gun.ones) {
+    if (bul.isDraw) {
+      bul.update();
+      ctx.drawImage(
+        bul.image, 0, 0,
+        bul.image.width, bul.image.height,
+        bul.x, bul.y,
+        bul.image.width, bul.image.height
+      );
+    }
+  }
+
+  for (const bomb of projector.ones) {
+    if (bomb.isDraw) {
+      bomb.update();
+      ctx.drawImage(
+        bomb.image, 0, 0,
+        bomb.image.width, bomb.image.height,
+        bomb.x, bomb.y,
+        bomb.image.width, bomb.image.height
+      );
+    }
+  }
+}
+
+function bonusUpdate() {
+  for (const bon of bonuses) {
+    if (bon.isDraw) {
+      ctx.drawImage(
+        bon.image, 0, 0,
+        bon.image.width, bon.image.height,
+        bon.x, bon.y,
+        bon.image.width, bon.image.height
+      );
+    }
+  }
+}
+
+function addZombi() {
+  const numb = randomInteger(0, 3);
+  const zombX = randomInteger(30, canvas.width - 2 * images.zombi.width);
+  const zombY = randomInteger(250, 400) * -1;
+  if (Player.level[numb]) {
+    Player.level[numb]--;
+    Player.level.count--;
+    return addZombiFunctions[numb](zombX, zombY);
+  }
+}
+
+function eggUpdate() {
+  for (const egg of eggs) {
+    if (egg.isDraw) {
+      egg.update();
+      ctx.drawImage(
+        egg.image, 0, 0,
+        egg.image.width, egg.image.height,
+        egg.x, egg.y,
+        egg.image.width, egg.image.height
+      );
+    }
+  }
+}
+
+function draw() {
+  ctx.clearRect(0, 0, canvas.width, canvas.height);
+  ctx.drawImage(
+    images.flat, 0, 0,
+    images.flat.width, images.flat.height,
+    0, 0,
+    canvas.width, canvas.height
+  );
+
+  const isLosed = collisionDetection();
+  if (isLosed) {
+    return;
+  }
+
+  bonusUpdate();
+  zombiUpdate();
+  weaponUpdate();
+  eggUpdate();
+
+  if (rightPressed && Player.x < canvas.width - Player.image.width) {
+    Player.x += speedPlayer;
+
+  } else if (leftPressed && Player.x > 0) {
+    Player.x -= speedPlayer;
+  }
```

Comment:
Both cases use `Player.x` and `speedPlayer`, so here we can change just +/- dependent on condition